src/controllers/v1/attachment/__tests__/index.test.ts

```diff
@@ -0,0 +1,304 @@
+          filename: 'doc.txt',
+          ipfs_hash: 'QmXVStDC6kTpVHY1shgBQmyA4SuSrYnNRnHSak5iB6Eehn',
+          size: 1024,
+        })
+      })
+
+      it('returns attachments', () => {
+        expect(response).to.deep.contain({
+          ...octetExample,
+          id: 'attachment-insert-test',
+        })
+      })
+    })
+
+    describe('also uploads JSON from req.body', () => {
+      beforeEach(async () => {
+        response = await controller.create({ body: { data: 'attachment-test' } } as any)
+      })
+
+      it('guploads to ipfs and gets an ipfs hash', () => {
+        expect(stubs.addFile.lastCall.args[0]).to.deep.contain({
+          filename: 'json',
+        })
+      })
+
+      it('does not update existing attachment', () => {
+        expect(stubs.update.callCount).to.equal(0)
+      })
+
+      it('persists a new attachment', () => {
+        expect(stubs.insert.lastCall.args[0]).to.equal('attachment')
+        expect(stubs.insert.lastCall.args[1]).to.deep.contain({
+          filename: 'json',
+          ipfs_hash: 'QmXVStDC6kTpVHY1shgBQmyA4SuSrYnNRnHSak5iB6Eehn',
+          size: 26,
+        })
+      })
+
+      it('returns attachments', () => {
+        expect(response).to.deep.contain({
+          filename: 'json',
+          size: 26,
+          id: 'attachment-insert-test',
+        })
+      })
+    })
+  })
+
+  describe('getAll() - GET /', () => {
+    beforeEach(async () => {
+      stubs.get.resetHistory()
+      stubs.get.callsFake(async () => [octetExample, jsonExample]), (response = await controller.getAll())
+    })
+
+    describe('if none found it', () => {
+      beforeEach(async () => {
+        stubs.get.callsFake(async () => [])
+        response = await controller.getAll()
+      })
+
+      it('returns an empty array', () => {
+        expect(response).to.be.empty
+      })
+    })
+
+    it('returns all attachments from a local db', () => {
+      expect(response.length).to.equal(2)
+      expect(response).to.deep.include.members([
+        {
+          filename: 'doc.txt',
+          size: 1024,
+          ipfs_hash: 'QmXVStDC6kTpVHY1shgBQmyA4SuSrYnNRnHSak5iB6Eehn',
+          id: 'attachment-1-test',
+          created_at: new Date('2020-10-10'),
+        },
+        {
+          id: 'attachment-2-test',
+          created_at: new Date('2024-01-05T09:39:29.785Z'),
+          filename: 'json',
+          ipfs_hash: 'QmXVStDC6kTpVHY1shgBQmyA4SuSrYnNRnHSak5iB6Eehn',
+          size: 26,
+        },
+      ])
+    })
+  })
+
+  describe('getById() - GET /{id}', () => {
+    describe('if attachment does not exist', () => {
+      beforeEach(async () => {
+        stubs.get.resetHistory()
+        stubs.update.resetHistory()
+        stubs.getFile.resetHistory()
+        stubs.get.callsFake(async () => [])
+
+        response = await controller
+          .getById({ headers: { accept: 'application/octet-stream' } } as any, 'non-existant-id')
```

Comment:
Maybe change `non-existant-id` to `non-existent-id` 

Edit: And maybe do the same change everywhere?